ui/common/StatusBar.tsx

```diff
@@ -14,25 +15,34 @@ export default function StatusBar({
   next,
   input,
   expected,
+  beginMsg,
   successMsg,
   inProgressMsg,
   errorMsg,
   full,
+  hints,
 }: {
   next: string
   input: string
   expected: string
+  beginMsg?: string
   successMsg?: string
   inProgressMsg?: string
   errorMsg?: string
   full?: boolean
+  hints?: boolean
 }) {
   const { activeView } = useLessonContext()
   const isActive = activeView === LessonView.Code
 
   const getStatus = () => {
-    if (!input) return Status.InProgress
+    if (!input) return Status.Begin
```

Comment:
Let's not use naked if's, they make the code less readable imo. I prefer something like this:
```ts
const getStatus = () => {
  if (!input) {
    return Status.Begin;
  }

  if (input === expected) {
    return Status.Success;
  }

  if (
    hints &&
    (!input || (expected.startsWith(input) && expected[0] === input[0]))
  ) {
    return Status.InProgress;
  }

  return Status.Error;
};

```